dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/secret/ServiceIntegrationResourceTest.java

```diff
@@ -0,0 +1,442 @@
+        Assert.assertNotNull(hosts1);
+        Assert.assertFalse(hosts1.isEmpty());
+        Assert.assertEquals(hosts1.get(0).getSiteId(),host.getIdentifier());
+        Assert.assertTrue(
+                hosts1.stream().anyMatch(hostView -> host.getIdentifier().equals(hostView.getSiteId()))
+                );
+
+        final Response detailedIntegrationResponse = serviceIntegrationResource.getDetailedServiceIntegration(request, response, serviceKey, host.getIdentifier());
+        Assert.assertEquals(HttpStatus.SC_OK, detailedIntegrationResponse.getStatus());
+        final ResponseEntityView responseEntityView3 = (ResponseEntityView) detailedIntegrationResponse.getEntity();
+        final ServiceIntegrationDetailedView serviceIntegrationDetailedView = (ServiceIntegrationDetailedView) responseEntityView3.getEntity();
+        Assert.assertEquals(serviceIntegrationDetailedView.getHost().getSiteId(),host.getIdentifier());
+
+        final Response deleteIntegrationsResponse = serviceIntegrationResource.deleteAllServiceIntegrationSecrets(request, response, serviceKey, host.getIdentifier());
+        Assert.assertEquals(HttpStatus.SC_OK, deleteIntegrationsResponse.getStatus());
+
+        final Response detailedIntegrationResponseAfterDelete = serviceIntegrationResource.getDetailedServiceIntegration(request, response, serviceKey, host.getIdentifier());
+        Assert.assertEquals(HttpStatus.SC_NOT_FOUND, detailedIntegrationResponseAfterDelete.getStatus());
+
+        //Now test the entry has been removed from the list of available configurations.
+        final Response hostIntegrationsResponseAfterDelete = serviceIntegrationResource.getServiceIntegrationByKey(request, response, serviceKey);
+        Assert.assertEquals(HttpStatus.SC_OK, hostIntegrationsResponseAfterDelete.getStatus());
+        final ResponseEntityView responseEntityViewAfterDelete = (ResponseEntityView) hostIntegrationsResponseAfterDelete.getEntity();
+        final ServiceIntegrationSiteView serviceIntegrationHostViewAfterDelete = (ServiceIntegrationSiteView) responseEntityViewAfterDelete.getEntity();
+        final ServiceIntegrationView serviceIntegrationViewAafterDelete = serviceIntegrationHostViewAfterDelete.getService();
+        Assert.assertEquals(0, serviceIntegrationViewAafterDelete.getConfigurationsCount());
+        Assert.assertEquals("lola", serviceIntegrationViewAafterDelete.getName());
+        final List<SiteView> expectedEmptyHosts = serviceIntegrationHostViewAfterDelete.getSites();
+        Assert.assertNotNull(expectedEmptyHosts);
+        Assert.assertTrue(expectedEmptyHosts.isEmpty());
+    }
+
+    @Test
+    public void  Test_Create_service_descriptor_Then_Create_Service_Integration_Then_Delete_One_Single_Secret() throws IOException {
+
+        final Map<String, Param> paramMap = ImmutableMap.of(
+                "param1", Param.newParam("val-1", false, Type.STRING, "label", "hint", true),
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "val-1" appears 4 times in this file; the first occurrence is on line 185](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)